catalog/core/catalog-core-validationparser/src/main/java/ddf/catalog/validation/impl/ValidationParser.java

```diff
@@ -203,6 +222,40 @@ private void parseValidators(Map<String, Object> root, Outer outer) {
         outer.validators = validators;
     }
 
+    @SuppressWarnings("unchecked")
+    private void parseMetacardValidators(@Nullable Map<String, Object> root, Outer outer) {
+        if (root == null || root.get(METACARD_VALIDATORS_PROPERTY) == null) {
+            return;
+        }
+
+        List<MetacardValidatorDefinition> metacardValidators = new ArrayList<>();
+        Object metacardValidatorsObj = root.get(METACARD_VALIDATORS_PROPERTY);
+        if (metacardValidatorsObj instanceof List) {
+            List<Map<String, Object>> metacardValidatorsList =
+                    (List<Map<String, Object>>) metacardValidatorsObj;
+            for (Map<String, Object> metacardTypeList : metacardValidatorsList) {
```

Comment:
Made a few modifications to do early return. I didn't split into multiple methods though.